### ISSUE TITLE
Extension does not acknowledge the existence of Oracle Linux

### ISSUE THREAD

**Issue by conzyor34** [chef-partners/azure-chef-extension#233]
This function inside `shared.sh` does not acknowledge the existence of Oracle Linux, so extension deployment fails for that platform.

```
get_linux_distributor(){
#### Using python -mplatform command to get distributor name #####
  if python -mplatform | grep centos > /dev/null; then
    linux_distributor='centos'
  elif python -mplatform | grep Ubuntu > /dev/null; then
    linux_distributor='ubuntu'
  elif python -mplatform | grep debian > /dev/null; then
    linux_distributor='debian'
  elif python -mplatform | grep redhat > /dev/null; then
    linux_distributor='rhel'
  fi
  echo "${linux_distributor}"
}
```
Here is the output for `python -mplatform` for Oracle Linux.

    # python -mplatform
    Linux-4.1.12-103.3.8.el7uek.x86_64-x86_64-with-oracle-7.4

This should be detected as `rhel` for the purposes of installing Chef.

No, I don't really want to be using Oracle Linux, neither should anybody else, but I don't have a choice in this case.

**Comment by bcolebank** [comment]
Trying to install Chef on my Oracle server and getting a similar behavior.  Log from the extension install at
/var/log/azure/Chef.Bootstrap.WindowsAzure.LinuxChefClient/1210.12.109.1004/CommandExecution.log:
2017/10/24 11:49:05.743178 INFO [Chef.Bootstrap.WindowsAzure.LinuxChefClient-121
0.12.109.1004] Target handler state: enabled
2017/10/24 11:49:05.743608 INFO [Chef.Bootstrap.WindowsAzure.LinuxChefClient-121
0.12.109.1004] [Enable] current handler state is: notinstalled
2017/10/24 11:49:05.767181 INFO [Chef.Bootstrap.WindowsAzure.LinuxChefClient-121
0.12.109.1004] Initialize extension directory
2017/10/24 11:49:05.767816 INFO [Chef.Bootstrap.WindowsAzure.LinuxChefClient-121
0.12.109.1004] Update settings file: 0.settings
2017/10/24 11:49:05.768395 INFO [Chef.Bootstrap.WindowsAzure.LinuxChefClient-121
0.12.109.1004] Install extension [install.sh]
2017/10/24 11:49:12.845411 INFO [Chef.Bootstrap.WindowsAzure.LinuxChefClient-121
0.12.109.1004] Remove extension handler directory: /var/lib/waagent/Chef.Bootstr
ap.WindowsAzure.LinuxChefClient-1210.12.109.1004

Contents of /var/log/azure/custom.log:
/var/lib/waagent/Chef.Bootstrap.WindowsAzure.LinuxChefClient-1210.12.109.1004
[Tue Oct 24 11:49:05 PDT 2017] Reading environment variables
[Tue Oct 24 11:49:05 PDT 2017] No environment variables found
Reading chef-client version from settings file
Call for Checking linux distributor
Chef-client is already installed
[Tue Oct 24 11:49:05 PDT 2017] Installing Azure Chef Extension gem
[Tue Oct 24 11:49:05 PDT 2017] Azure Chef Extension gem installation failed
/var/lib/waagent/linxchef
/var/lib/waagent/linxchef

Was able to bootstrap it separately, but would much rather use the extension as part of the deployment.

**Comment by piyushawasthi** [comment]
Fixes: https://github.com/chef-partners/azure-chef-extension/pull/236

**Comment by NimishaS** [comment]
This issue has been resolved in the latest release `1210.12.110.1000`.
